decompiler/pipeline/controlflowanalysis/variable_name_generation.py

```diff
@@ -72,26 +33,34 @@ class RenamingScheme(ABC):
     """Base class for different Renaming schemes."""
 
     def __init__(self, task: DecompilerTask) -> None:
-        """Collets all needed variables for renaming + filters already renamed + function arguments out"""
-        collector = VariableCollector(task._ast.condition_map)
+        """Collets all needed variables for renaming + filters which should not be renamed"""
+        collector = VariableCollector()
         collector.visit_ast(task._ast)
+        self._ast = task._ast
         self._params: List[Variable] = task._function_parameters
-        self._loop_vars : List[Variable] = collector.get_loop_variables()
-        self._variables: List[Variable] = list(filter(self._filter_variables, collector.get_variables()))
+        self._variables: Set[Variable] = set(filter(self._filter_variables, collector.get_variables()))
         
 
     def _filter_variables(self, item: Variable) -> bool:
         """Return False if variable is either a:
             - parameter
-            - renamed loop variable
             - GlobalVariable
         """
-        return not item in self._params and not (item in self._loop_vars and item.name.find("var_") == -1) and not isinstance(item, GlobalVariable)
+        return not item in self._params and not isinstance(item, GlobalVariable)
+
+
+    def renameVariables(self):
+        """Rename all collected variables with a naming scheme."""
+        names: dict[Variable, Variable] = {}
+        for var in self._variables:
+            names[var] = var.copy(self.getVariableName(var))
```

Comment:
Another minority, I think for clarification it could help to write here `var.copy(name=self.getVariableName(var))`.

Reply:
Sure